projects/wpdx/src/app/common-components/photo-gallery/photo-gallery.component.html

```diff
@@ -0,0 +1,13 @@
+
+<div *ngIf='photoList.length > 0' class='photo-gallery'>
+    <div *ngFor='let photo of photoList; let i = index'>
+        <div *ngIf='photo.photo_lnk?true:removePhoto(i)' [class.current-photo]='i === currentPhotoIndex' [class.left]='i < currentPhotoIndex' [class.right]='i > currentPhotoIndex'>
```

Comment:
it's a pretty bad idea to have a boolean expression with side-effects, as well as removing items from a list while iterating over it.
Also, there's no need to remove these items from the list, as we might need the history later somewhere else - simply filter the list in code before rendering the element.